Sources/Bot/Services/DispatchService.swift

```diff
@@ -0,0 +1,198 @@
+import Foundation
+import Result
+import ReactiveSwift
+import ReactiveFeedback
+
+public final class DispatchService {
+    private let integrationLabel: PullRequest.Label
+    private let topPriorityLabels: [PullRequest.Label]
+    private let requiresAllStatusChecks: Bool
+    private let statusChecksTimeout: TimeInterval
+
+    private let logger: LoggerProtocol
+    private let gitHubAPI: GitHubAPIProtocol
+    private let scheduler: DateScheduler
+
+    private var mergeServices: [String: MergeService]
+    private let mergeServiceLifecycleObserver: Signal<MergeServiceLifecycleEvent, NoError>.Observer?
+
+    public let healthcheck: Healthcheck
+
+    public init(
+        integrationLabel: PullRequest.Label,
+        topPriorityLabels: [PullRequest.Label],
+        requiresAllStatusChecks: Bool,
+        statusChecksTimeout: TimeInterval,
+        logger: LoggerProtocol,
+        gitHubAPI: GitHubAPIProtocol,
+        gitHubEvents: GitHubEventsServiceProtocol,
+        scheduler: DateScheduler = QueueScheduler(),
+        mergeServiceLifecycleObserver: Signal<MergeServiceLifecycleEvent, NoError>.Observer? = nil
+    ) {
+        self.integrationLabel = integrationLabel
+        self.topPriorityLabels = topPriorityLabels
+        self.requiresAllStatusChecks = requiresAllStatusChecks
+        self.statusChecksTimeout = statusChecksTimeout
+
+        self.logger = logger
+        self.gitHubAPI = gitHubAPI
+        self.scheduler = scheduler
+
+        self.mergeServices = [:]
+        self.mergeServiceLifecycleObserver = mergeServiceLifecycleObserver
+
+        healthcheck = Healthcheck(scheduler: scheduler)
+
+        gitHubAPI.fetchPullRequests()
+            .flatMapError { _ in .value([]) }
+            .map { pullRequests in
+                pullRequests.filter { $0.isLabelled(with: self.integrationLabel) }
+            }
+            .observe(on: scheduler)
+            .startWithValues { pullRequests in
+                self.dispatchInitial(pullRequests: pullRequests)
+            }
+
+        gitHubEvents.events
+            .observe(on: scheduler)
+            .observeValues { [weak self] gitHubEvent in
+                switch gitHubEvent {
+                case let .pullRequest(event):
+                    self?.pullRequestDidChange(event: event)
+                case let .status(event):
+                    self?.statusChecksDidChange(event: event)
+                case .ping:
+                    break
+                }
+            }
+    }
+
+    private func dispatchInitial(pullRequests: [PullRequest]) {
+        let dispatchTable = Dictionary(grouping: pullRequests) { pullRequest in pullRequest.target.ref }
```

Comment:
[nit] even though we prefer not to use unnamed parameters, in this case it's quite obvious that each item of `pullRequests` is `pullRequest`, so I'd shorten it

```suggestion
        let dispatchTable = Dictionary(grouping: pullRequests) { $0.target.ref }
```